packages/xstate-wallet/src/ui/application-workflow.tsx

```diff
@@ -44,7 +46,7 @@ export const ApplicationWorkflow = (props: Props) => {
       <Heading textAlign="center" mb={0}>
         {messages[getApplicationStateValue(current)]}
       </Heading>
-      {!isConfirmCreateChannel(current) && (
+      {!isConfirmCreateChannel(current) && !isApplicationChallenging(current) && (
```

Comment:
Should `isConfirmCreateChannel` ever return `true` when `isApplicationChallenging` returns `true`?

What seems to be going on here is:
- the UI is always in exactly one "display mode"
- we have `isApplicationX` boolean functions defined in `./selectors'`
- so, we should probably have a `applicationDisplayMode` which returns an enum/union type
- we display the UI based on the return value of `applicationDisplayMode`